File: mobile/lib/models/habit.dart
class Habit {
  final String id;
  final String badHabit;
  final String cueTrigger;
  final String replacementHabit;
  final String reward;
  final String frequency;
  final String scheduledTime;
  final String category;
  final bool isActive;

  Habit({
    required this.id,
    required this.badHabit,
    required this.cueTrigger,
    required this.replacementHabit,
    required this.reward,
    this.frequency = 'daily',
    this.scheduledTime = '22:30',
    this.category = 'Health & Diet',
    this.isActive = true,
  });

  factory Habit.fromJson(Map<String, dynamic> json) {
    return Habit(
      id: json['id'] ?? '',
      badHabit: json['bad_habit'] ?? '',
      cueTrigger: json['cue_trigger'] ?? '',
      replacementHabit: json['replacement_habit'] ?? '',
      reward: json['reward'] ?? '',
      frequency: json['frequency'] ?? 'daily',
      scheduledTime: json['scheduled_time'] ?? '22:30',
      category: json['category'] ?? 'General',
      isActive: json['is_active'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'bad_habit': badHabit,
      'cue_trigger': cueTrigger,
      'replacement_habit': replacementHabit,
      'reward': reward,
      'frequency': frequency,
      'scheduled_time': scheduledTime,
      'category': category,
      'is_active': isActive,
    };
  }
}

class CatalogItem {
  final String id;
  final String category;
  final String title;
  final String description;
  final String icon;

  CatalogItem({
    required this.id,
    required this.category,
    required this.title,
    required this.description,
    this.icon = 'sparkles',
  });
}

File: services/habit-service/internal/domain/habit.go
package domain

import "time"

// Habit represents the core habit substitution entity
type Habit struct {
	ID                   string    `json:"id"`
	UserID               string    `json:"user_id"`
	BadHabit             string    `json:"bad_habit"`
	Frequency            string    `json:"frequency"`
	ScheduledTime        string    `json:"scheduled_time"`
	CueTrigger           string    `json:"cue_trigger"`
	ReplacementHabit     string    `json:"replacement_habit"`
	Reward               string    `json:"reward"`
	Category             string    `json:"category"`
	PreferredWindowStart string    `json:"preferred_window_start"` // e.g. "06:00"
	PreferredWindowEnd   string    `json:"preferred_window_end"`   // e.g. "10:00"
	IsActive             bool      `json:"is_active"`
	CreatedAt            time.Time `json:"created_at"`
	UpdatedAt            time.Time `json:"updated_at"`
}

// CreateHabitDTO defines the payload required to construct a new Habit
type CreateHabitDTO struct {
	UserID               string `json:"user_id"`
	UserTier             string `json:"user_tier"`
	BadHabit             string `json:"bad_habit"`
	Frequency            string `json:"frequency"`
	ScheduledTime        string `json:"scheduled_time"`
	CueTrigger           string `json:"cue_trigger"`
	ReplacementHabit     string `json:"replacement_habit"`
	Reward               string `json:"reward"`
	Category             string `json:"category"`
	PreferredWindowStart string `json:"preferred_window_start"`
	PreferredWindowEnd   string `json:"preferred_window_end"`
}

// UpdateHabitTimeDTO contains data needed to reschedule a habit
type UpdateHabitTimeDTO struct {
	HabitID       string `json:"habit_id"`
	ScheduledTime string `json:"scheduled_time"`
}
